lib/Support/parser.cpp

```diff
@@ -0,0 +1,238 @@
+	}
+}
+
+enum comp {
+	lessThan, greaterThan, equall, error, accept
+};
+comp precedanceTable[7][7];
+
+void buildOperatorPrecedenceTable() {
+    precedanceTable[0][0] = greaterThan;
+	precedanceTable[0][1] = greaterThan;
+	precedanceTable[0][2] = greaterThan;
+	precedanceTable[0][3] = lessThan;
+	precedanceTable[0][4] = greaterThan;
+	precedanceTable[0][5] = lessThan;
+	precedanceTable[0][6] = greaterThan;
+
+    precedanceTable[1][0] = lessThan;
+	precedanceTable[1][1] = greaterThan;
+	precedanceTable[1][2] = greaterThan;
+	precedanceTable[1][3] = lessThan;
+	precedanceTable[1][4] = greaterThan;
+	precedanceTable[1][5] = lessThan;
+	precedanceTable[1][6] = greaterThan;
+
+    precedanceTable[2][0] = lessThan;
+	precedanceTable[2][1] = lessThan;
+	precedanceTable[2][2] = greaterThan;
+	precedanceTable[2][3] = lessThan;
+	precedanceTable[2][4] = greaterThan;
+	precedanceTable[2][5] = lessThan;
+	precedanceTable[2][6] = greaterThan;
+
+    precedanceTable[3][0] = greaterThan;
+	precedanceTable[3][1] = lessThan;
+	precedanceTable[3][2] = lessThan;
+	precedanceTable[3][3] = lessThan;
+	precedanceTable[3][4] = equall;
+	precedanceTable[3][5] = lessThan;
+	precedanceTable[3][6] = error;
+
+    precedanceTable[4][0] = greaterThan;
+	precedanceTable[4][1] = greaterThan;
+	precedanceTable[4][2] = greaterThan;
+	precedanceTable[4][3] = error;
+	precedanceTable[4][4] = greaterThan;
+	precedanceTable[4][5] = error;
+	precedanceTable[4][6] = greaterThan;
+
+    precedanceTable[5][0] = error;
+	precedanceTable[5][1] = greaterThan;
+	precedanceTable[5][2] = greaterThan;
+	precedanceTable[5][3] = error;
+	precedanceTable[5][4] = greaterThan;
+	precedanceTable[5][5] = error;
+	precedanceTable[5][6] = greaterThan;
+
+    precedanceTable[6][0] = lessThan;
+	precedanceTable[6][1] = lessThan;
+	precedanceTable[6][2] = lessThan;
+	precedanceTable[6][3] = lessThan;
+	precedanceTable[6][4] = error;
+	precedanceTable[6][5] = lessThan;
+	precedanceTable[6][6] = accept;
+
+}
+
+BoolExpression * constructNodeOperator(StackNode* operate, StackNode* s1, StackNode*s2){
+    if(operate!= NULL && s1!=NULL && s2!=NULL){
+        Token t = operate->term;
+        expressionType oo;
+        if(t.token_type == AND_TOKEN){
+            oo=AND;
+        }else if(t.token_type == OR_TOKEN){
+            oo=OR;
+        }
+        BoolExpression* e1 = s1->expr;
+	    BoolExpression* e2 = s2->expr;
+        return new BoolExpression(oo, e1, e2);
+    }else{
+        syntax_error();
+		return new BoolExpression(END);
+    }
+}
+
+BoolExpression * constructNodeNegator(StackNode* s1){
+    if(s1!=NULL){
+        BoolExpression* e1 = s1->expr;
+        return new BoolExpression(NOT, nullptr, e1);
+    }else{
+        syntax_error();
+		return new BoolExpression(END);
+    }
+}
+
+//turing a SingleCond from term to expr
+StackNode* TERM_TO_EXP(StackNode* s) {	
+    if(s!=NULL){
+	    return new StackNode(EXPR, new BoolExpression(VARIABLE, stoi(s->term.lexeme)));
+	}else{
+		syntax_error();
```

Comment:
These errors either need to be an `assert` (if `s` should never be `nullptr` in any correct use of the API) or be forwarded back to the caller in some way if it can be triggered by incorrect user input.